File: frontends/main/src/app/configurator/services/api/form-model.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {FormModel} from '@core/model/form-model';

@Injectable({
	providedIn: 'root'
})
export class FormModelService {
	constructor(private http: HttpClient) {}

	getFormModels(projectId: string): Observable<FormModel[]> {
		return this.http.get<FormModel[]>(`/api/superuser/configurator/projects/${projectId}/config/form-models`);
	}

	getFormModel(projectId: string, formModelId: string): Observable<FormModel> {
		return this.http.get<FormModel>(`/api/superuser/configurator/projects/${projectId}/config/form-models/${formModelId}`);
	}

	createFormModel(projectId: string, formModel: FormModel): Observable<FormModel> {
		return this.http.post<FormModel>(`/api/superuser/configurator/projects/${projectId}/config/form-models`, formModel);
	}

	updateFormModel(projectId: string, formModelId: string, formModel: FormModel): Observable<FormModel> {
		return this.http.put<FormModel>(`/api/superuser/configurator/projects/${projectId}/config/form-models/${formModelId}`, formModel);
	}

	deleteFormModel(projectId: string, formModelId: string): Observable<void> {
		return this.http.delete<void>(`/api/superuser/configurator/projects/${projectId}/config/form-models/${formModelId}`);
	}
}

File: backend/src/main/java/ch/rodano/core/services/bll/configurator/FormModelService.java
package ch.rodano.core.services.bll.configurator;

import java.util.List;
import java.util.UUID;

import ch.rodano.api.config.FormModelDTO;

public interface FormModelService {

	List<FormModelDTO> getFormModels(UUID projectId);

	FormModelDTO getFormModel(UUID projectId, UUID formModelId);

	FormModelDTO createFormModel(UUID projectId, FormModelDTO formModel);

	FormModelDTO updateFormModel(UUID projectId, UUID formModelId, FormModelDTO formModel);

	void deleteFormModel(UUID projectId, UUID formModelId);
}
